libs/flexIPC/fleXdIPCCommon/src/FleXdIPCMsg.cpp

```diff
@@ -0,0 +1,466 @@
+            {
+                return m_value3;
+            }
+            void FleXdIPCAdtHdr::setValue_3(uint32_t value)
+            {
+                m_value3 = value;
+                m_headerMask |= IPC_MSG_ADDITIONAL_HEADER_3_FLAG;
+            }
+            void FleXdIPCAdtHdr::resetValue_3()
+            {
+                m_value3 = 0;
+                if (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_3_FLAG) m_headerMask ^= IPC_MSG_ADDITIONAL_HEADER_3_FLAG;
+            }
+
+            const uint32_t FleXdIPCAdtHdr::getValue_4() const
+            {
+                return m_value4;
+            }
+            void FleXdIPCAdtHdr::setValue_4(uint32_t value)
+            {
+                m_value4 = value;
+                m_headerMask |= IPC_MSG_ADDITIONAL_HEADER_4_FLAG;
+            }
+            void FleXdIPCAdtHdr::resetValue_4()
+            {
+                m_value4 = 0;
+                if (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_4_FLAG) m_headerMask ^= IPC_MSG_ADDITIONAL_HEADER_4_FLAG;
+            }
+
+            const uint32_t FleXdIPCAdtHdr::getValue_5() const
+            {
+                return m_value5;
+            }
+            void FleXdIPCAdtHdr::setValue_5(uint32_t value)
+            {
+                m_value5 = value;
+                m_headerMask |= IPC_MSG_ADDITIONAL_HEADER_5_FLAG;
+            }
+            void FleXdIPCAdtHdr::resetValue_5()
+            {
+                m_value5 = 0;
+                if (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_5_FLAG) m_headerMask ^= IPC_MSG_ADDITIONAL_HEADER_5_FLAG;
+            }
+
+            const uint8_t FleXdIPCAdtHdr::getNextHeaderMask() const
+            {
+                return m_next != nullptr ? m_next->getMask() : 0;
+            }
+
+            const uint16_t FleXdIPCAdtHdr::getSize() const
+            {
+                return (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_0_FLAG ? sizeof(m_value0) : 0) +
+                    (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_1_FLAG ? sizeof(m_value1) : 0) +
+                    (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_2_FLAG ? sizeof(m_value2) : 0) +
+                    (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_3_FLAG ? sizeof(m_value3) : 0) +
+                    (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_4_FLAG ? sizeof(m_value4) : 0) +
+                    (m_headerMask & IPC_MSG_ADDITIONAL_HEADER_5_FLAG ? sizeof(m_value5) : 0) +
+                    (m_next != nullptr ? sizeof(IPC_MSG_ADDITIONAL_HEADER_NEXT_HEADER_MASK_BIT_COUNT) + m_next->getSize() : 0);
+            }
+
+            FleXdIPCAdtHdr* FleXdIPCAdtHdr::createNext()
+            {
+                if (m_next == nullptr)
+                {
+                    m_next = new FleXdIPCAdtHdr();
+                    m_headerMask |= IPC_MSG_ADDITIONAL_HEADER_NEXT_HEADER_MASK_FLAG;
+                }
+                return m_next;
+            }
+
+            void FleXdIPCAdtHdr::deleteNext()
+            {
+                if (m_next != nullptr)
```

Comment:
Ditto. 'if (m_next != nullptr)' is fairly useless since standard defines the behaviour of 'delete' well when 'nullptr' is supplied.